language_detector/main.py

```diff
@@ -1,9 +1,16 @@
 """This is the entry point of the program."""
-
+# coding=utf-8
 from .languages import LANGUAGES
 
 
 def detect_language(text, languages=LANGUAGES):
     """Returns the detected language of given text."""
-    # implement your solution here
-    pass
+    spanish, german, english = [set(LANGUAGES[x]['common_words']) for x in range(3)]
+    language_list = [spanish, german, english]
```

Comment:
This works perfectly for these inputs. Nice work. Really like your coding style.
Was just wondering when it comes to larger inputs, hashtable can have a very high space complexity, and how to resolve this challenge?